src/libraries/System.Private.CoreLib/src/System/SpanHelpers.ByteMemOps.cs

```diff
@@ -0,0 +1,538 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if TARGET_AMD64 || TARGET_ARM64 || (TARGET_32BIT && !TARGET_ARM) || TARGET_LOONGARCH64
+// JIT is guaranteed to unroll blocks up to 64 bytes in size
+#define HAS_CUSTOM_BLOCKS
+#endif
+
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal static partial class SpanHelpers // .ByteMemOps
+    {
+#if TARGET_ARM64 || TARGET_LOONGARCH64
+        private const ulong MemmoveNativeThreshold = ulong.MaxValue;
+#elif TARGET_ARM
+        private const nuint MemmoveNativeThreshold = 512;
+#else
+        private const nuint MemmoveNativeThreshold = 2048;
+#endif
+        // TODO: Determine optimal value
```

Comment:
```suggestion
```
Anything TODO here?

Reply:
um.. do you mean it needs a link? I didn't spent enough time testing this treshold on different cpus/platforms yet. I presume it also can't be too big to avoid regressing NAOT compiled for generic cpu.

Reply:
I wasn't able to find any regressions from that infinite threshold on modern hw in microbenchmarks, but it's very possible the picture is different in real world due to those non-temporal stores etc. 
I agree it won't hurt to change that treshold to some big value like 32kb etc

Reply:
I'll remove this comment as part of the `GT_STORE_DYN_BLK` clean up PR to avoid spinning CI for comment change if you don't mind